t/t2035-checkout-locked.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/sh
+
+test_description='checkout must be able to overwrite open files'
+. ./test-lib.sh
+
+test_expect_success 'setup' '
+
+	echo hello >world &&
+	mkdir dear &&
+	echo hello>dear/world &&
+	git add -A &&
+	git commit -m hi &&
+	git tag hi &&
+	git checkout -b bye &&
+	git mv dear cruel &&
+	echo goodbye >world &&
+	echo goodbye >cruel/world &&
+	git add -A &&
+	git commit -m bye
+'
+
+test_expect_success 'lock for read ' '
+
+	git checkout bye &&
+	exec 7<world &&
+	exec 8<cruel/world &&
```

Comment:
Opening one file handle should be enough to reproduce. If opening two file handles or opening a file in a subdirectory is interesting for you to test, please add a separate test for it (with an appropriate test message. Having minimal tests and testing incrementally makes test failures easier to understand and diagnose.

Reply:
Done